### PR TITLE
deploy diffs for formplayer

### PR DESCRIPTION
**Pull request by snopoke**
[USH-705](https://dimagi-dev.atlassian.net/browse/USH-705)

* Refactoring deploy code to create cleaner separation between commcare and fomplayer
* Formplayer deploy diff output (see below)
* Formplayer release tagging
* Formplayer deploy success email with diff output (see below)
* Deploy event on DD

### console output
image

### email  output
image

## QA

With the exception of the Sentry release creation, I have tested this for both formplayer and commcare on staging, with and without github credentials.

### CODE REVIEW

File: src/commcare_cloud/commands/deploy/command.py
Comment (snopoke) [inline src/commcare_cloud/commands/deploy/command.py:1]: mostly just moved code from `deploy.py` after extracting out the commcare deploy bits into `commcare.py`

File: src/commcare_cloud/commands/deploy/commcare.py
Comment (snopoke) [inline src/commcare_cloud/commands/deploy/commcare.py:1]: mostly just moved code from `deploy.py`

File: src/commcare_cloud/fab/git_repo.py
Comment (snopoke) [inline src/commcare_cloud/fab/git_repo.py:1]: this code is just moved from `fab/utils.py`

File: src/commcare_cloud/fab/diff_templates/email.html.j2
Comment (snopoke) [inline src/commcare_cloud/fab/diff_templates/email.html.j2:9]: Mostly copied from https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqadmin/templates/hqadmin/partials/project_snapshot.html

**Review by millerdev — COMMENTED** [review]
Haven't gotten through reviewing the full PR yet, but had one comment so far.

File: src/commcare_cloud/commands/deploy/formplayer.py
Comment (esoergel) [inline src/commcare_cloud/commands/deploy/formplayer.py]: Auth is used for two things, I believe:
1. To obtain write permissions to set a deploy tag in the repo
2. To avoid being rate limited by github when making a bunch of requests for PR info.

Auth is not needed for other github operations, like fetching the latest commit.

We might consider separating these concerns, or at least being more explicit about what we require when, since this will likely be run by users who don't need or want github auth, those who do want auth for full deploy diffs, and Dimagi users who also have write access to create tags.  I'm not certain we handle all those situations correctly, and it seems prone to confusion.

File: src/commcare_cloud/commands/deploy/formplayer.py
Comment (esoergel) [inline src/commcare_cloud/commands/deploy/formplayer.py:111]: Think this should be a hard requirement?  This is consistent with existing behavior, so a change would be out-of-scope for this PR, but I could see us refusing to deploy a `tag_deploy_commits` env unless you provide auth (to prevent silent deploys that don't show up properly).

**Review by esoergel — APPROVED** [review]

File: src/commcare_cloud/fab/deploy_diff.py
Comment (esoergel) [reply src/commcare_cloud/fab/deploy_diff.py]: (resolved in a subsequent commit)

File: src/commcare_cloud/fab/deploy_diff.py
Comment (esoergel) [reply src/commcare_cloud/fab/deploy_diff.py]: Nevermind, I see this referenced separately in the template

File: src/commcare_cloud/commands/deploy/formplayer.py
Comment (snopoke) [reply src/commcare_cloud/commands/deploy/formplayer.py:89]: 689c4111f

File: src/commcare_cloud/commands/deploy/formplayer.py
Comment (snopoke) [reply src/commcare_cloud/commands/deploy/formplayer.py:19]: I agree it's not ideal but I'm leaving it out of scope for this PR since it requires more significant changes to the API for getting a GitHub repo. Also considering @esoergel comment [here](https://github.com/dimagi/commcare-cloud/pull/4623#discussion_r609074897) I think we should consider in more broard github auth rework.

Comment (millerdev) [reply src/commcare_cloud/commands/deploy/formplayer.py:19]: I could not tell if it was safe to reference `fabric.state.env` outside of a fabric context, but it seems important to determine if it is safe for this PR since you are introducing a new workflow where formplayer deploy (done with ansible) is using things from HQ deploy (done with fabric) that reference the fabric context. It looks like the only attribute accessed in `commcare_cloud.fab.git_repo` is `tag_deploy_commits`, but I'm uncertain of how that value is calculated.

**Comment by millerdev** [comment]
Approved if the fabric reference in formplayer deploy (ansible context) is safe.

File: src/commcare_cloud/commands/deploy/formplayer.py
Comment (snopoke) [reply src/commcare_cloud/commands/deploy/formplayer.py:19]: `fabric.state.env` is just a global dictionary (that supports attribute access). It is safe to use as a normal python object.

File: src/commcare_cloud/commands/utils.py
Comment (millerdev) [reply src/commcare_cloud/commands/utils.py:53]: Forgot that I asked for tests here. It would be nice if they were added.

Comment (snopoke) [reply src/commcare_cloud/commands/utils.py:53]: Thanks for the reminder. https://github.com/dimagi/commcare-cloud/pull/4667

**Pull request merged**
